packages/browser/src/core/events/interfaces.ts

```diff
@@ -1,229 +1,35 @@
-   * - page: https://segment.com/docs/connections/spec/page/#properties
-   * - screen: https://segment.com/docs/connections/spec/screen/#properties
-   * @example
-   *  { artistID: 2435325, songID: 13532532 }
-   */
-  properties?: EventProperties
-  /**
-   * An object literal representing traits
-   * - identify: https://segment.com/docs/connections/spec/identify/#traits
-   * - group: https://segment.com/docs/connections/spec/group/#traits
-   * @example
-   *  { name: "john", age: 25 }
-   */
-  traits?: Traits
-
-  integrations?: Integrations
-  context?: AnalyticsContext | Options
-  options?: Options
-
-  userId?: ID
-  anonymousId?: ID
-  groupId?: ID
-  previousId?: ID
-
-  event?: string
-
-  /**
-   * {@link https://github.com/segmentio/analytics.js-integrations/blob/2d5c637c022d2661c23449aed237d0d546bf062d/integrations/segmentio/lib/index.js#L284}
-   */
-  writeKey?: string
-
-  /**
-   *  {@link https://github.com/segmentio/analytics.js-integrations/blob/2d5c637c022d2661c23449aed237d0d546bf062d/integrations/segmentio/lib/index.js#L151}
-   */
-  sentAt?: Date
-
-  /**
-   * {@link https://github.com/segmentio/analytics.js-integrations/blob/2d5c637c022d2661c23449aed237d0d546bf062d/integrations/segmentio/lib/index.js#L311-L320}
-   */
-  _metadata?: {
-    failedInitializations?: unknown[]
-    bundled?: string[]
-    unbundled?: string[]
-    nodeVersion?: string
-    bundledConfigIds?: string[]
-    unbundledConfigIds?: string[]
-    bundledIds?: string[]
-  }
-
-  timestamp?: Date | string
-}
-
-/**
- * A Plan allows users to specify events and which destinations they would like them to be sent to
- */
-export interface Plan {
-  track?: TrackPlan
-  identify?: TrackPlan
-  group?: TrackPlan
-}
-
-export interface TrackPlan {
-  [key: string]: PlanEvent | undefined
-  // __default SHOULD always exist, but marking as optional for extra safety.
-  __default?: PlanEvent
-}
-
-export interface PlanEvent {
-  /**
-   * Whether or not this plan event is enabled
-   */
-  enabled: boolean
-  /**
-   * Which integrations the plan event applies to
-   */
-  integrations?: {
-    [key: string]: boolean
-  }
-}
-
-export interface ReservedTraits {
-  address: Partial<{
-    city: string
-    country: string
-    postalCode: string
-    state: string
-    street: string
-  }>
-  age: number
-  avatar: string
-  birthday: Date
-  company: Partial<{
-    name: string
-    id: string | number
-    industry: string
-    employee_count: number
-  }>
-  plan: string
-  createdAt: Date
-  description: string
-  email: string
-  firstName: string
-  gender: string
-  id: string
-  lastName: string
-  name: string
-  phone: string
-  title: string
-  username: string
-  website: string
+export interface SegmentEvent extends CoreSegmentEvent {}
+
+export type {
+  Integrations,
+  Plan,
+  TrackPlan,
+  PlanEvent,
+  Callback,
+  JSONArray,
+  JSONValue,
+  JSONPrimitive,
+  JSONObject,
```

Comment:
To make this PR reviewable, I'm try to avoid changing the file structure too much. Thus, I'm simply exporting a lot of types that were previously declared here. Elsewhere I do the same thing.